internal/postgres/config.go

```diff
@@ -299,9 +284,18 @@ done
 		naming.ReplicationCACertPath,
 	)
 
+	// this is used to close out the while loop started above after adding the required
+	// auto grow annotation scripts
+	finalDone := `done
+`
+
 	// Elide the above script from `ps` and `top` by wrapping it in a function
 	// and calling that.
-	wrapper := `monitor() {` + script + `}; export -f monitor; exec -a "$0" bash -ceu monitor`
+	wrapper := `monitor() {` +
+		script + // main script body
+		util.AutogrowAnnotationScript("pgdata") + // pgdata annotation script
```

Comment:
what do you think about making this a bash function instead of go?

something like

```bash
check_resize() {
  directory=$1

  // do stuff with $directory
  ...
}

// original cert rotation stuff

check_resize "pgdata"

if [ -d /pgwal ]; then
  check_resize "pgwal"
fi
```

Reply:
That might be a bit easier to understand, plus it will avoid changes in another package. I'll see what I can get working.